libs/core/langchain_core/schema/chat_message_history.py

```diff
@@ -0,0 +1,69 @@
+from abc import ABC, abstractmethod
+from typing import List, Union
+
+from langchain_core.messages import (
+    AIMessage,
+    BaseMessage,
+    HumanMessage,
+)
+
+
+class ChatMessageHistoryInterface(ABC):
+    """Interface for storing chat message history.
+
+    See `ChatMessageHistory` for default implementation.
+    The default implementation is the BaseChatMessageHistory class.
+
+    Example:
+        .. code-block:: python
+
+            class FileChatMessageHistory(BaseChatMessageHistory):
+                storage_path:  str
+                session_id: str
+
+               @property
+               def messages(self):
+                   with open(os.path.join(storage_path, session_id), 'r:utf-8') as f:
+                       messages = json.loads(f.read())
+                    return messages_from_dict(messages)
+
+               def add_message(self, message: BaseMessage) -> None:
+                   messages = self.messages.append(_message_to_dict(message))
+                   with open(os.path.join(storage_path, session_id), 'w') as f:
+                       json.dump(f, messages)
+
+               def clear(self):
+                   with open(os.path.join(storage_path, session_id), 'w') as f:
+                       f.write("[]")
+    """
+
+    messages: List[BaseMessage]
+    """A list of Messages stored in-memory."""
+
+    @abstractmethod
+    def add_user_message(self, message: Union[HumanMessage, str]) -> None:
```

Comment:
* add_user_message, add_ai_message should not exist

@abstactmethod should be to `add_messages` which supports bulk load.

`add_message` so should exist and be implemented by default to use `add_messages`

Here's a PR to the existing base class that attempts to fix some of the issues: https://github.com/langchain-ai/langchain/pull/15709/files

Unfortunately, for backwards compatibility the implementation of add_message needs to be more complex and actually account for the fact that most classes don't have add_messages